### PR TITLE
crawl: 0.29 -> 0.30.1

### PR DESCRIPTION
**Pull request by FurstPlattner**
Adjusted for changes in build process, updated patch, and ran nixpkgs-fmt over the files.

###### Description of changes

Updated to 0.30.1: [Release notes 0.30](https://github.com/crawl/crawl/releases/tag/0.30.0), [0.30.1](https://github.com/crawl/crawl/releases/tag/0.30.1). This involved a minor update to the patch file and some change in how the .desktop files are deposited. I'm not really very knowledgeable about these things, but they seem to work.

Note that I didn't run nixpkgs-review because apparently my laptop does not have sufficient memory.

I think this could safely be backported, but I'm not sure how to add the labels for it.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by catern** [comment]
This fails to build on my system, with the following error:

```
during RTL pass: ira
mon-cast.cc: In function 'void mons_cast(monster*, bolt, spell_type, mon_spell_slot_flags, bool)':
mon-cast.cc:6577:1: internal compiler error: Segmentation fault
 6577 | }
      | ^
c++ -std=c++11 -O2 -pipe     -Wall -Wformat-security -Wundef -Wextra -Wno-missing-field-initializers -Wno-implicit-fallthrough -Wno-type-limits -Wno-uninitialized  -Wno-array-bounds -Wno-format-zero-length -Wmissing-declarations -Wredundant-decls -Wno-parentheses -Wwrite-strings -Wshadow -pedantic -Wuninitialized -Iutil -I. -isystem /nix/store/zbpbkp6svg6szmq30q547ghdr13zafja-lua-5.1.5/include -isystem /nix/store/k2pb1jvavk4ljxlx2mymyc04sng3gjc1-sqlite-3.42.0-dev/include   -DWIZARD -DASSERTS  -DCLUA_BINDINGS -DSAVE_DIR_PATH=\"~/.crawl\" -DDATA_DIR_PATH=\"/nix/store/3jxxc0w2fm48zbir5p1a4ni4gr9whprz-crawl-0.30.1/\" -D_DEFAULT_SOURCE -D_XOPEN_SOURCE=600 -isystem /nix/store/685px7fmxjygvzi7gj450hsn4h7rx1kj-ncurses-6.4-dev/include -MMD -c status.cc -o status.o
0x1baca57 diagnostic_impl(rich_location*, diagnostic_metadata const*, int, char const*, __va_list_tag (*) [1], diagnostic_t)
	???:0
0x1bad8a7 internal_error(char const*, ...)
	???:0
0xddc31f crash_signal(int)
	???:0
0xc1b58d ira_tune_allocno_costs()
	???:0
Please submit a full bug report, with preprocessed source (by using -freport-bug).
Please include the complete backtrace with any bug report.
See <https://gcc.gnu.org/bugs/> for instructions.
c++ -std=c++11 -O2 -pipe     -Wall -Wformat-security -Wundef -Wextra -Wno-missing-field-initializers -Wno-implicit-fallthrough -Wno-type-limits -Wno-uninitialized  -Wno-array-bounds -Wno-format-zero-length -Wmissing-declarations -Wredundant-decls -Wno-parentheses -Wwrite-strings -Wshadow -pedantic -Wuninitialized -Iutil -I. -isystem /nix/store/zbpbkp6svg6szmq30q547ghdr13zafja-lua-5.1.5/include -isystem /nix/store/k2pb1jvavk4ljxlx2mymyc04sng3gjc1-sqlite-3.42.0-dev/include   -DWIZARD -DASSERTS  -DCLUA_BINDINGS -DSAVE_DIR_PATH=\"~/.crawl\" -DDATA_DIR_PATH=\"/nix/store/3jxxc0w2fm48zbir5p1a4ni4gr9whprz-crawl-0.30.1/\" -D_DEFAULT_SOURCE -D_XOPEN_SOURCE=600 -isystem /nix/store/685px7fmxjygvzi7gj450hsn4h7rx1kj-ncurses-6.4-dev/include -MMD -c stepdown.cc -o stepdown.o
```

I wonder if this is because I'm on a recent CPU revision? I'm on a i9-13900K

File: pkgs/games/crawl/default.nix [hunk 1]
```diff
@@ -1,77 +1,118 @@
...
+, freetype
+, pngcrush
+, advancecomp
+, tileMode ? false
+, enableSound ? tileMode
+, buildPackages
 
-# MacOS / Darwin builds
+  # MacOS / Darwin builds
 , darwin ? null
```
Comment (AndersonTorres) [inline pkgs/games/crawl/default.nix:27]: Null arguments should be avoided.
```suggestion
 , darwin
```

File: pkgs/games/crawl/default.nix [hunk 2]
```diff
@@ -1,77 +1,118 @@
...
-                  with darwin.apple_sdk.frameworks; [
-                    AppKit AudioUnit CoreAudio ForceFeedback Carbon IOKit OpenGL
-                   ]
-                ));
+  buildInputs = [ lua5_1 ncurses sqlite zlib ]
+    ++ (with python3.pkgs; [ pyyaml ])
+    ++ lib.optionals tileMode [ freetype libGL libGLU libpng SDL2 SDL2_image  ]
+    ++ lib.optional enableSound SDL2_mixer
+    ++ (lib.optionals stdenv.isDarwin (
+    assert (lib.assertMsg (darwin != null) "Must have darwin frameworks available for darwin builds");
```
Comment (AndersonTorres) [inline pkgs/games/crawl/default.nix:60]:
```suggestion
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/games/crawl/default.nix [hunk 3]
```diff
@@ -1,77 +1,117 @@
...
+, freetype
+, pngcrush
+, advancecomp
+, tileMode ? false
+, enableSound ? tileMode
+, buildPackages
 
-# MacOS / Darwin builds
-, darwin ? null
+  # MacOS / Darwin builds
```
Comment (AndersonTorres) [inline pkgs/games/crawl/default.nix:26]:
```suggestion
```

File: pkgs/games/crawl/default.nix [hunk 4]
```diff
@@ -1,77 +1,117 @@
...
+  ++ lib.optional enableSound "SOUND=y";
 
   postInstall = ''
-    ${lib.optionalString tileMode "mv $out/bin/crawl $out/bin/crawl-tiles"}
-    sed -i 's#/usr/games/##' debian/crawl${lib.optionalString tileMode "-tiles"}.desktop
-    install -m 444 -D debian/crawl${lib.optionalString tileMode "-tiles"}.desktop \
-      $out/share/applications/crawl${lib.optionalString tileMode "-tiles"}.desktop
-    install -m 444 -D dat/tiles/stone_soup_icon-512x512.png $out/share/icons/hicolor/512x512/apps/crawl.png
+    make install-xdg-data ${builtins.concatStringsSep " " finalAttrs.makeFlags}
+    ${lib.optionalString tileMode ''
```
Comment (AndersonTorres) [inline pkgs/games/crawl/default.nix:95]: Use `+` instead of interpolation.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/games/crawl/crawl_purify.patch [hunk 5]
```diff
@@ -1,28 +1,21 @@
...
 +++ b/crawl-ref/source/util/find_font
-@@ -1,6 +1,6 @@
- #! /bin/sh
- 
+@@ -7,7 +7,7 @@
+ # some hard-coded font directories to check. Note: if you are using
+ # font-config with other directories, you don't need to edit this -- that case
+ # is handled by the `fc-list` call below.
 -FONTDIRS="/usr/share/fonts /usr/local/share/fonts /usr/*/lib/X11/fonts"
 +FONTDIRS="${fontsPath}/share/fonts"
```
Comment (AndersonTorres) [inline pkgs/games/crawl/crawl_purify.patch:10]: \Does it  work?

Comment (FurstPlattner) [reply pkgs/games/crawl/crawl_purify.patch:10]: While I haven't played extensively, all game text I've encountered renders correctly on my machine.

If you mean "Does the weird bash script this is patching do what it is intended to do", then I don't know.

Comment (AndersonTorres) [reply pkgs/games/crawl/crawl_purify.patch:10]: You should.

Usually we use `@var@` plus substituteAll.

**Comment by wegank** [comment]
Closed by #282456.

**Pull request closed without merging**